applications/FluidDynamicsApplication/custom_utilities/estimate_dt_utilities.cpp

```diff
@@ -0,0 +1,165 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//                   Ruben Zorrilla
+//
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "utilities/element_size_calculator.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "estimate_dt_utilities.h"
+
+
+namespace Kratos
+{
+
+    void EstimateDtUtility::SetCFL(const double CFL)
+    {
+        mCFL = CFL;
+    }
+
+    void EstimateDtUtility::SetDtMin(const double DtMin)
+    {
+        mDtMin = DtMin;
+    }
+
+    void EstimateDtUtility::SetDtMax(const double DtMax)
+    {
+        mDtMax = DtMax;
+    }
+
+    double EstimateDtUtility::EstimateDt() const
+    {
+        KRATOS_TRY;
+
+        // Get the projected element size function according to the corresponding geometry
+        // Note that in here it is assumed that all the elements in the model part feature the same geometry
+        const auto& r_geom = mrModelPart.ElementsBegin()->GetGeometry();
+        ElementSizeFunctionType projected_h_func = EstimateDtUtility::GetMinimumElementSizeFunction(r_geom);
+
+        // Obtain the maximum CFL
+        const double current_dt = mrModelPart.GetProcessInfo().GetValue(DELTA_TIME);
+        const double current_cfl = block_for_each<MaxReduction<double>>(mrModelPart.Elements(), [&](Element& rElement) -> double {
+            return CalculateElementCFL(rElement, projected_h_func, current_dt);
+        });
```

Comment:
so no need for the TLS now? (don't remember the details from my branch)

@peterjwilson FYI this is an example of a simple max reduction :)

To me this looks so nice!

Reply:
Nope. The TLS was required in order to avoid the `GeometryUtils` allocations within the parallel loops. As we got rid of that it is no longer required.